libs/core/langchain_core/utils/pydantic.py

```diff
@@ -12,3 +17,35 @@ def get_pydantic_major_version() -> int:
 
 
 PYDANTIC_MAJOR_VERSION = get_pydantic_major_version()
+
+
+# How to type hint this?
+def pre_init(func: Callable) -> Any:
+    """Decorator to run a function before model initialization."""
+
+    @root_validator(pre=True)
```

Comment:
if this changes everything to pre=True, won't this create problems with values[x] not existing, where before it would?

Reply:
nvm I believe you